src/App.js

```diff
@@ -12,17 +12,25 @@ class App extends React.Component {
   }
 
   inputSubmit = (value) => {
-      this.setState(prevState => ({
-          list: [...prevState.list, value]
-      }));
+    this.setState(prevState => ({
+        list: [...prevState.list, value = {id: new Date().getTime(), text: value}]
+    }));
+  }
+
+  deleteItem = (deleteId) => {
+    this.setState(prevState => ({
+      list: [...prevState.list.filter( (listItem) => {return listItem.id !== deleteId;} )]
```

Comment:
- no need to put it inside the array and spread, just filter the `prevstate.list `and assign it to` list`
- no need to wrap inside curly brackets and return with `return` ketword
- one parameter no need to be wrapped inside of parentheses 

```suggestion
      list: prevState.list.filter(listItem => listItem.id !== deleteId)
```